clients/lsp-rust.el

```diff
@@ -1340,12 +1342,21 @@ meaning."
                    (pos (lsp--position-to-point position))
                    (overlay (make-overlay pos pos nil 'front-advance 'end-advance)))
              (let ((concat-label
-                    (string-join
-                     (mapcar
-                      (lambda (label)
-                        (when (lsp-structure-p label)
-                          (gethash "value" label "")))
-                      label))))
+                    (cl-typecase label
+                      (vector
+                       (string-join
+                        (mapcar
+                         (lambda (label)
+                           (when (lsp-structure-p label)
+                             (gethash "value" label "")))
+                         label)))
+                      (string
+                       label)
+                      (t
+                       (unless lsp-rust-analyzer-already-warned-about-inlay-hint-type
+                         (message "Unexpected type for inlay hint: %s" (type-of label))
+                         (setq lsp-rust-analyzer-already-warned-about-inlay-hint-type t))
+                       ""))))
```

Comment:
Do we really need this ? Isn't it enough to just ignore this case ?

Reply:
I like the idea that it'll print out at least some message.

Reply:
Yeah, that was my thought too.  Silent failure is really annoying to figure out.